main/accounts/Account/index.ts

```diff
@@ -193,15 +193,32 @@ class FrameAccount {
     if (knownRequest) {
       if (knownRequest.res) {
         const { id, jsonrpc } = req.payload || {}
-        
         knownRequest.res({ id, jsonrpc, result })
       }
-      store.navClearReq(req.handlerId)
-      delete this.requests[req.handlerId]
-      this.update()
+
+      this.clearRequest(knownRequest)
     }
   }
 
+  rejectRequest (req: AccountRequest, error: EVMError) {
+    const knownRequest = this.requests[req.handlerId]
+
+    if (knownRequest) {
+      if (knownRequest.res) {
+        const { id, jsonrpc } = req.payload || {}
```

Comment:
you're not using req directly so can destructure it above like 
`rejectRequest ({ handlerId, payload = {} }: AccountRequest, error: EVMError) {`

Reply:
:+1: